File: examples/chapter08/blurimage.m
%%
% We use an image that is built into MATLAB.
load mandrill
[m,n] = size(X)
image(X), colormap(gray(256))
axis equal, title('Original image')    % ignore this line

%%
% We define the one-dimensional tridiagonal blurring matrices.
v = [1/4 1/2 1/4];
B = spdiags( repmat(v,m,1), -1:1, m,m);
C = spdiags( repmat(v,n,1), -1:1, n,n);

%%
% Finally, we show the results of using $k=12$ repetitions of the blur in
% each direction.
blur = @(X) B^12 * X * C^12;
image(blur(X))
axis equal, title('Blurred image')    % ignore this line

File: julia/scripts/chapter08/blurimage.jl

using Images,TestImages
img = testimage("mandrill");
@show m,n = size(img)

using Plots
plot(Gray.(img),title="Original image",aspect_ratio=1)

X = @. Float64(Gray(img))

using SparseArrays
B = spdiagm(0=>fill(0.5,m),1=>fill(0.25,m-1),-1=>fill(0.25,m-1));
C = spdiagm(0=>fill(0.5,n),1=>fill(0.25,n-1),-1=>fill(0.25,n-1));

blur = X -> B^12 * X * C^12;
plot(Gray.(blur(X)),title="Blurred image",aspect_ratio=1)
